packages/cli/test/unit/commands/login.test.ts

```diff
@@ -5,22 +5,45 @@ import { useUser } from '../../mocks/user';
 describe('login', () => {
   it('should not allow the `--token` flag', async () => {
     client.setArgv('login', '--token', 'foo');
-    const exitCode = await login(client);
-    expect(exitCode).toEqual(2);
-    expect(client.outputBuffer).toEqual(
+    const exitCodePromise = login(client);
+    await expect(client.stderr).toOutput(
       'Error! `--token` may not be used with the "login" command\n'
     );
+    await expect(exitCodePromise).resolves.toEqual(2);
   });
 
   it('should allow login via email as argument', async () => {
     const user = useUser();
     client.setArgv('login', user.email);
-    const exitCode = await login(client);
-    expect(exitCode).toEqual(0);
-    expect(
-      client.outputBuffer.includes(
+    const exitCodePromise = login(client);
+    await expect(client.stderr).toOutput(
+      `Success! Email authentication complete for ${user.email}`
+    );
+    await expect(exitCodePromise).resolves.toEqual(0);
+  });
+
+  describe('interactive', () => {
+    it('should allow login via email', async () => {
+      const user = useUser();
+      client.setArgv('login');
+      const exitCodePromise = login(client);
+      await expect(client.stderr).toOutput(`> Log in to Vercel`);
+
+      // Move down to "Email" option
+      client.stdin.write('\x1B[B'); // Down arrow
+      client.stdin.write('\x1B[B'); // Down arrow
+      client.stdin.write('\x1B[B'); // Down arrow
+      client.stdin.write('\r'); // Return key
```

Comment:
Maybe we can make this kind of interaction easier in a future PR. I'm certainly not going to remember the proper code for "down arrow" for example.